### ISSUE TITLE
Valid query not possible to execute

### ISSUE THREAD

**Issue by svrcekmichal** [Youshido/GraphQL#49]
This query combining unions and fragments does not work:
```
{
  user(id: 2) {
    ...user
  }
}

fragment user on UserUnion {
  ... on User {
    id
  }
}
```
schema contains query `user` which can return `UserUnion` which will be resolved to type `User` with field `id`. Exception thrown: `Fragment reference "user" not found on model "User"`

**Comment by viniychuk** [comment]
Hi @svrcekmichal , I'm trying to reproduce your situation here and it seems like it works on my side. Would you be able to make a test that fails so we can work against it?

Here's my working example:

```php

$userType = new ObjectType([
    'name'   => 'User',
    'fields' => [
        'id'   => new IntType(),
        'name' => new StringType(),
    ]
]);

$userUnion = new UnionType([
    'name' => 'UserUnion',
    'types' => [$userType],
    'resolveType' => function() use ($userType) {
        return $userType;
    }
]);


$processor = new Processor(new Schema([
    'query' => new ObjectType([
        'name'   => 'RootQueryType',
        'fields' => [
            'user' => [
                'type'    => $userUnion,
                'resolve' => function () {
                    return [
                        'id'   => 1,
                        'name' => 'alex'
                    ];
                }
            ]
        ]
    ])
]));

$processor->processPayload('{ user { ...userInfo } } fragment userInfo on User { id, name }');
echo json_encode($processor->getResponseData()) . "\n";
```

**Comment by svrcekmichal** [comment]
@viniychuk You have bad query processed. Fragment should query `UserUnion` and have inline query on `User`. This is the query with your schema you should process:
`{ user { ...userInfo } } fragment userInfo on UserUnion { ... on User { id } }`

**Comment by viniychuk** [comment]
@svrcekmichal yep. And because of this scenario I'm now doing some performance improvements along the way :)

**Comment by svrcekmichal** [comment]
@viniychuk Great to hear that. I didn't find any performance bottleneck for now, but you probably know about few of them. Btw. is it possible to define fields in some sort of callback? In some cases it would rapidly reduce schema size if only types and fields needed would be initialized.

**Comment by viniychuk** [comment]
Schema cachine mechanism will be there fairly soon, but it will be possible only after we create "validated schema cache", otherwise it's hard to guarantee consistent state and behavior.

**Comment by viniychuk** [comment]
fixed in v1.2.9.1
